src/js/jsx/sections/style/Radius.jsx

```diff
@@ -40,72 +40,130 @@ define(function (require, exports, module) {
     var Radius = React.createClass({
         mixins: [FluxMixin, Coalesce],
 
+        /**
+         * Keeps track of the radius on Scrub start
+         *
+         * @type {number}
+         */
+        _initialScrubRadius: null,
+
         propTypes: {
             document: React.PropTypes.object.isRequired
         },
 
-        shouldComponentUpdate: function (nextProps) {
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return this.state !== nextState;
+        },
+
+        componentWillReceiveProps: function (nextProps) {
             var getRelevantProps = function (props) {
-                var layers = props.document.layers.selected;
+                    var layers = props.document.layers.selected.filter(function (layer) {
+                        return layer.radii;
+                    });
 
-                return collection.pluckAll(layers, ["id", "bounds", "radii"]);
-            };
+                    return collection.pluckAll(layers, ["id", "bounds", "radii"]);
+                },
+                relevantProps = getRelevantProps(this.props),
+                nextRelevantProps = getRelevantProps(nextProps);
+
+            if (!Immutable.is(relevantProps, nextRelevantProps)) {
+                this._calculateState(nextProps);
+            }
+        },
+
+        componentWillMount: function () {
+            this._calculateState(this.props);
+        },
+
+        /**
+         * Given the props object, calculates and sets the state
+         * This is refactored out of componentWillReceiveProps, because 
+         * it doesn't run on initial render, causing a null state
+         *
+         * @private
+         * @param {object} props
+         */
+        _calculateState: function (props) {
+            var layers = props.document.layers.selected.filter(function (layer) {
+                    return layer.radii;
+                }),
+                scalars = layers.map(function (layer) {
+                    return layer.radii.scalar || 0;
+                }),
+                maxRadius = layers
+                    .toSeq()
+                    .filter(function (layer) {
+                        return !!layer.bounds;
+                    })
+                    .reduce(function (sides, layer) {
+                        return sides.concat(Immutable.List.of(layer.bounds.width / 2, layer.bounds.height / 2));
+                    }, Immutable.List())
+                    .min();
+
+            this.setState({
+                layers: layers,
+                scalars: scalars,
+                maxRadius: maxRadius,
+                maxRadiusInput: Math.floor(maxRadius)
+            });
+        },
+
+        _handleScrubStart: function () {
+            this._initialScrubRadius = collection.uniformValue(this.state.scalars);
```

Comment:
I feel very strongly that _initialScrubRadius should be a property of the state.  `uniformValue` is not hard to compute, and it is fundamentally a stateful property of the component.